src/Module/Conversation/Channel.php

```diff
@@ -0,0 +1,438 @@
+
+			if (Feature::isEnabled($this->session->getLocalUserId(), 'trending_tags')) {
+				$this->page['aside'] .= TrendingTags::getHTML(self::$content);
+			}
+
+			// We need the editor here to be able to reshare an item.
+			$o .= $this->conversation->statusEditor([], 0, true);
+		}
+
+		$items = $this->getItems($request, $this->session, $this->cache);
+
+		if (!$this->database->isResult($items)) {
+			$this->systemMessages->addNotice($this->l10n->t('No results.'));
+			return $o;
+		}
+
+		$o .= $this->conversation->render($items, Conversation::MODE_CHANNEL, false, false, 'created', $this->session->getLocalUserId());
+
+		$pager = new BoundariesPager(
+			$this->l10n,
+			$this->args->getQueryString(),
+			$items[0]['created'],
+			$items[count($items) - 1]['created'],
+			self::$itemsPerPage
+		);
+
+		if ($this->pConfig->get($this->session->getLocalUserId(), 'system', 'infinite_scroll')) {
+			$o .= HTML::scrollLoader();
+		} else {
+			$o .= $pager->renderMinimal(count($items));
+		}
+
+		return $o;
+	}
+
+	/**
+	 * Computes module parameters from the request and local configuration
+	 *
+	 * @throws HTTPException\BadRequestException
+	 * @throws HTTPException\ForbiddenException
+	 */
+	protected function parseRequest(array $request)
+	{
+		self::$accountTypeString = $request['accounttype'] ?? $this->parameters['accounttype'] ?? '';
+		self::$accountType       = User::getAccountTypeByString(self::$accountTypeString);
+
+		self::$content = $this->parameters['content'] ?? '';
+		if (!self::$content) {
+			self::$content = self::FORYOU;
+		}
+
+		if (!in_array(self::$content, [self::WHATSHOT, self::FORYOU, self::FOLLOWERS, self::IMAGE, self::VIDEO, self::AUDIO])) {
+			throw new HTTPException\BadRequestException($this->l10n->t('Channel not available.'));
+		}
+
+		if ($this->mode->isMobile()) {
+			self::$itemsPerPage = $this->pConfig->get(
+				$this->session->getLocalUserId(),
+				'system',
+				'itemspage_mobile_network',
+				$this->config->get('system', 'itemspage_network_mobile')
+			);
+		} else {
+			self::$itemsPerPage = $this->pConfig->get(
+				$this->session->getLocalUserId(),
+				'system',
+				'itemspage_network',
+				$this->config->get('system', 'itemspage_network')
+			);
+		}
+
+		if (!empty($request['item'])) {
+			$item          = Post::selectFirst(['parent-uri-id'], ['id' => $request['item']]);
+			self::$item_id = $item['parent-uri-id'] ?? 0;
+		} else {
+			self::$item_id = 0;
+		}
+
+		self::$min_id = $request['min_id']       ?? null;
+		self::$max_id = $request['max_id']       ?? null;
+		self::$max_id = $request['last_created'] ?? self::$max_id;
+	}
+
+	/**
+	 * Computes the displayed items.
+	 *
+	 * Community pages have a restriction on how many successive posts by the same author can show on any given page,
+	 * so we may have to retrieve more content beyond the first query
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	protected function getItems(array $request, UserSession $session, ICanCache $cache)
```

Comment:
Both `$session` and `$cache` references can be replaced in this method respectively by `$this->session` and `$this->cache` as they are available in this method's scope.

Reply:
Changed